lib/mac/bindings.js

```diff
@@ -283,7 +287,7 @@ NobleBindings.prototype._onPeripheralDiscover = function(peripheral, advertiseme
     localName: advertisementData.localName,
     txPowerLevel: advertisementData.txPowerLevel,
     manufacturerData: advertisementData.manufacturerData,
-    serviceData: advertisementData.serviceData,
+    serviceData: advertisementData.serviceData ? advertisementData.serviceData : [],
```

Comment:
There's some inconsistencies yosemite.js and node-core-bluetooth here:
 * https://github.com/sandeepmistry/node-core-bluetooth/blob/master/lib/central-manager-delegate.js#L68
 * https://github.com/sandeepmistry/noble/blob/master/lib/mac/yosemite.js#L140

One is an object, the other is an array, but it should be an array. Unrelated to this PR, but we need to correct node-core-bluetooth.

Then we don't need this check?